lib/did_you_mean/extra_features/key_error.rb

```diff
@@ -0,0 +1,25 @@
+module DidYouMean
+  module ExtraFeatures
+    module KeyErrorReciver
+      def fetch(name, *args, &block)
+        super
+      rescue KeyError => e
+        this = self
+        e.define_singleton_method(:receiver) { this }
+        e.define_singleton_method(:name) { name }
+        e.define_singleton_method(:args) { nil }
+        raise e
+      end
+    end
+    Hash.prepend KeyErrorReciver
+
+    class KeyNameChecker < MethodNameChecker
+      def candidates
+        { @method_name.inspect => @receiver.keys.map(&:inspect) }
```

Comment:
The `map` call should be moved out of this method since it would make correction results inaccurate. Instead, override the `#corrections` method like this:

```ruby
def corrections
  super.map(&:inspect)
end
```